package NSnippets.NProto;

message TProperty {
    optional bytes Key = 1;
    optional bytes Value = 2;
    optional bool IsMain = 3;
}

message TImage {
    optional bytes Url = 1;
}

message TPreviewItem {
    optional bytes Name = 1;
    optional bytes Description = 2;
    optional bytes Url = 3;
    repeated TImage Images = 4;
    repeated TProperty Properties = 5;
}

message TRawPreview {
    repeated TPreviewItem PreviewItems = 1;
    optional uint32 ResultsCountInAll = 2;
    optional bytes Template = 3;
}
